docs/spec/spec.md

```diff
@@ -0,0 +1,332 @@
+# Specification: Ballerina TCP Library
+
+_Owners_: @shafreenAnfar @bhashinee  
+_Reviewers_: @shafreenAnfar  
+_Created_: 2021/12/20  
+_Updated_: 2021/12/20  
+_Issue_: [#2171](https://github.com/ballerina-platform/ballerina-standard-library/issues/2171)
+
+# Introduction
+
+This is the specification for TCP standard library which is used to implement TCP compliant `listener` and `client` using [Ballerina programming language](https://ballerina.io/), which is an open-source programming language for the
+cloud that makes it easier to use, combine, and create network services. 
+
+
+# Contents
+1. [Overview](#1-overview)
+2. [Listener](#2-listener)
+    * 2.1. [Configurations](#21-configurations)
+    * 2.2. [Initialization](#22-initialization)
+3. [Service Types](#3-service-types)
+    * 3.1. [Service](#31-service)
+    * 3.2. [Connection Service](#32-connection-service)
+        * 3.2.1. [Remote methods associated with the Connection Service](#321-remote-methods-associated-with-the-connection-service)
+            * [onBytes](#onbytes)
+            * [onError](#onerror)
+            * [onClose](#onclose)
+4. [Client](#4-client)
+    * 4.1. [Configurations](#41-configurations)
+    * 4.2. [Initialization](#42-initialization)
+    * 4.3. [Send and receive data](#43-send-and-receive-data)
+        * [writeBytes](#writebytes)
+        * [readBytes](#readbytes)
+        * [close](#close)
+5. [Securing the TCP Connections](#5-securing-the-tcp-connections)
+    * 5.1 [Using the TLS protocol](#51-using-the-tls-protocol)
+6. [Samples](#6-samples)
+
+## 1. [Overview](#1-overview)
+
+TCP is a protocol that enables applications to exchange messages over a network. It is designed to ensure the successful delivery of data over the network. This specification elaborates on how Ballerina language provides a tested TCP client and server implementation that is compliant with the [RFC 793](https://datatracker.ietf.org/doc/html/rfc793).
+
+## 2. [Listener](#2-listener)
+
+The `tcp:Listener` is used to listen to the incoming socket request. It can be constructed with a port number and optionally providing other configurations. When initiating the listener it opens up the port and attaches the `tcp:service`. 
```

Comment:
```suggestion
The `tcp:Listener` is used to listen to the incoming socket request. It can be constructed with a port number and optionally providing other configurations. When initiating the listener it opens up the port and attaches the `tcp:Service`. 
```